README.md

```diff
@@ -121,3 +121,74 @@ may have the following optional properties in addition to the standard
   `this.inputPaths`. (The name `treeDir` is for historical reasons.)
 - `line`: Line in which the error occurred (one-indexed)
 - `column`: Column in which the error occurred (zero-indexed)
+
+### `Plugin.prototype.input`
+
+Plugin now provides a easy way to read from input path via `Plugin.prototype.input`. `input` provides file operation methods to read file, directory etc. `input` automatically attaches the corresponding inputpath to the filename. `input` searches the filename from left to right and the right most file takes the precedance if there are two files exists with same path from the root in different inputpaths.
```

Comment:
Let's decouple this paragraph a bit, here is an idea:


------

### `Plugin.prototype.input`

An api which enables a plugin to easily read from one or more input directories ergonomically and safely.

More specifically:
* `this.input` reads from the provided `inputPaths`. No path concatenation required
* `this.input` provides readOnly file system APIs. This prevents a plugin from erroneously mutating its inputs.
* `this.input` provides a `merged` view of inputs, this allows every plugin to easily support multiple inputs, without the use of `broccoli-merge-trees` or implementing a complex merge algorithm.
* `this.input.at(index)` provides access to each individual input if desired.

Examples:


```js
// old
fs.readFileSync(this.inputPaths[0] + '/file.txt');

// new (merged): Most Common
this.input.readFileSync('file.txt');

// new (indexed): For when you need to disambiguate between inputs.
this.input.at(0).readFileSync('file.txt);
```

```js
// ReadOnly
this.input.writeFileSync // throws error
```

```js
// Merged
...
```

#### List of Methods

* list
* of 
* methods
* on 
* input

Reply:
cc @rwjblue / @thoov  / @SparshithNR